### PR TITLE
New process, step 1 : company selection

### PR DESCRIPTION
**Pull request by ArnoHolo**

### CODE REVIEW

File: app/controllers/api/diagnoses_controller.rb [hunk 1]
```diff
@@ -6,6 +6,17 @@ def show
       @diagnosis = Diagnosis.find params[:id]
     end
 
+    def create
+      facility = UseCases::SearchFacility.with_siret_and_save params[:siret]
+      visit = Visit.create advisor: current_user, facility: facility if facility
+      diagnosis = Diagnosis.new visit: visit if visit
+      if facility && visit && diagnosis.save
+        render json: { id: diagnosis.id }, status: :created, location: api_diagnosis_url(diagnosis)
```
Comment (Apemb) [inline app/controllers/api/diagnoses_controller.rb]: why not use jbuilder like the other api controllers ?

File: app/controllers/companies_controller.rb [hunk 2]
```diff
@@ -1,18 +1,10 @@
...
 
 class CompaniesController < ApplicationController
-  def search_by_siret
-    company = UseCases::SearchCompany.with_siret params[:siret]
-    @company_name = ApiEntrepriseService.company_name(company)
-    location_hash = company['etablissement_siege']['commune_implantation']
-    @company_location = "#{location_hash['code']} #{location_hash['value']}" if location_hash
-  end
-
   def search_by_siren
```
Comment (Apemb) [inline app/controllers/companies_controller.rb]: Still useful ?

File: app/services/api_entreprise_service.rb [hunk 3]
```diff
@@ -17,9 +17,19 @@ def fetch_facility_with_siret(siret)
     end
 
     def company_name(company_json)
-      company_name = company_json['entreprise']['nom_commercial']
-      company_name = company_json['entreprise']['raison_sociale'] if company_name.blank?
-      company_name
+      entreprise_hash = company_json.dig('entreprise')
+      return nil unless entreprise_hash
+      company_name = entreprise_hash.dig('nom_commercial')
+      company_name = entreprise_hash.dig('raison_sociale') if company_name.blank?
```
Comment (Apemb) [inline app/services/api_entreprise_service.rb]: why not use the same guard pattern as above ? `return nil unless entreprise_hash`

File: app/controllers/companies_controller.rb (hunk 2 above)
Comment (ArnoHolo) [inline app/controllers/companies_controller.rb]: Still used by _add_by_company_modal, but soon to be deleted!

File: app/services/api_entreprise_service.rb (hunk 3 above)
Comment (ArnoHolo) [inline app/services/api_entreprise_service.rb]: It's not the same case, I want `company_name = entreprise_hash.dig('raison_sociale')` to be called only if `entreprise_hash.dig('nom_commercial')` returned a blank string.
If you're speaking of l24, any other way to write this seems more complicated / less understandable.

**Pull request merged**